doc/whats-new.rst

```diff
@@ -335,6 +335,8 @@ New Features
 - Added ability to save ``DataArray`` objects directly to Zarr using :py:meth:`~xarray.DataArray.to_zarr`.
   (:issue:`7692`, :pull:`7693`) .
   By `Joe Hamman <https://github.com/jhamman>`_.
+- Many error messages related to invalid dimensions or coordinates now always show the list of valid dims/coords (:pull:`8079`).
```

Comment:
Woops, I added this under April for some reason. But there is no entry for the next release. Also, does this belong under "new features"?

Reply:
> Woops, I added this under April for some reason.

That's okay, whether or not the whatsnew entries are in the right place gets checked and corrected before we [release a new version of xarray](https://github.com/pydata/xarray/blob/main/HOW_TO_RELEASE.md).

> Also, does this belong under "new features"?

I would put it under "internals" maybe?

Reply:
I thought putting it under features because it's kind of user-facing, but yeah neither feels entirely correct. Both are fine for me.

Reply:
I moved this under "internals", I found examples of similar "improve error messages" PRs variously under new features, bugfixes and internals.